configcatclient/user.py

```diff
@@ -24,10 +29,27 @@ def get_attribute(self, attribute):
         return self.__custom.get(attribute) if self.__custom else None
 
     def __str__(self):
-        dump = {
-            'Identifier': self.__identifier,
-            'Email': self.__data.get('Email'),
-            'Country': self.__data.get('Country'),
-            'Custom': self.__custom,
-        }
-        return json.dumps(dump, indent=4)
+        dump = OrderedDict([
+            ('Identifier', self.__identifier),
+            ('Email', self.__data.get('Email')),
+            ('Country', self.__data.get('Country'))
+        ])
+        if self.__custom:
+            dump.update(self.__custom)
+
+        filtered_dump = OrderedDict([(k, v) for k, v in dump.items() if v is not None])
+        return json.dumps(filtered_dump, separators=(',', ':'))
+
+    @staticmethod
+    def attribute_value_from_datetime(value):
+        if value is None or not isinstance(value, datetime):
+            return None
```

Comment:
We have a minor inconsistency with the JS SDK here: https://github.com/configcat/common-js/blob/5247ed626ab1bb484fcd90c0fdf4cb485e60e3d5/src/User.ts#L13-L15

Enforcing the correct type is the main goals of these helpers, so I think an early error (exception) would be a better choice here. But either way, we'd need to have a consistent behavior accross our SDKs.

Reply:
After discussing this, we came to the conclusion that it's preferable not to crash the user application with an exception in the case of an invalid input type, so let's leave this as it is now.